File: csharp/ToolGood.Algorithm/Operands/OperandBoolean.cs
using ToolGood.Algorithm.Enums;

namespace ToolGood.Algorithm
{
	internal sealed class OperandBoolean : Operand
	{
		private readonly bool _value;
		public OperandBoolean(bool obj)
		{
			_value = obj;
		}
		public override bool IsBoolean => true;
		public override OperandType Type => OperandType.BOOLEAN;
		public override bool BooleanValue => _value;

		public override Operand ToNumber(string errorMessage) { return _value ? One : Zero; }
		public override Operand ToNumber(string errorMessage, params object[] args) { return _value ? One : Zero; }

		public override Operand ToBoolean(string errorMessage) { return this; }
		public override Operand ToBoolean(string errorMessage, params object[] args) { return this; }

		public override Operand ToText(string errorMessage) { return Create(_value ? "TRUE" : "FALSE"); }
		public override Operand ToText(string errorMessage, params object[] args) { return Create(_value ? "TRUE" : "FALSE"); }

		public override string ToString() { return _value ? "true" : "false"; }
	}
}

File: java/src/main/java/toolgood/algorithm/operands/OperandBoolean.java
package toolgood.algorithm.operands;

import toolgood.algorithm.Operand;
import toolgood.algorithm.enums.OperandType;

public final class OperandBoolean extends Operand {
    private final boolean _value;

    public OperandBoolean(boolean obj) {
        _value = obj;
    }

    @Override
    public boolean IsBoolean() { return true; }

    @Override
    public OperandType Type() { return OperandType.BOOLEAN; }

    @Override
    public boolean BooleanValue() { return _value; }

    @Override
    public Operand ToNumber(String errorMessage) { return _value ? One : Zero; }

    @Override
    public Operand ToNumber(String errorMessage, Object[] args) { return _value ? One : Zero; }

    @Override
    public Operand ToBoolean(String errorMessage) { return this; }

    @Override
    public Operand ToBoolean(String errorMessage, Object[] args) { return this; }

    @Override
    public Operand ToText(String errorMessage) { return Create(_value ? "TRUE" : "FALSE"); }

    @Override
    public Operand ToText(String errorMessage, Object[] args) { return Create(_value ? "TRUE" : "FALSE"); }

    @Override
    public String toString() { return _value ? "true" : "false"; }
}
